### PR TITLE
feat: ignore performance counters with partial aggregation

### PR DESCRIPTION
**Pull request by rahulguptajss**
PR with bool variable in Instance struct

### CODE REVIEW

File: cmd/collectors/zapiperf/zapiperf.go [hunk 1]
```diff
@@ -345,6 +346,23 @@ func (z *ZapiPerf) loadParamInt(name string, defaultValue int) int {
...
 
+func (z *ZapiPerf) isPartialAggregation(instance *node.Node) bool {
+	aggregation := instance.GetChildS("aggregation")
+	if aggregation != nil {
+		aggregationData := aggregation.GetChildS("aggregation-data")
+		if aggregationData != nil {
+			result := aggregationData.GetChildS("result")
+			if result != nil {
+				r := result.GetContentS()
+				if r == "partial_aggregation" {
```
Comment (cgrinds) [inline cmd/collectors/zapiperf/zapiperf.go:357]: return r == "partial_aggregation"

File: cmd/collectors/zapiperf/zapiperf_test.go [hunk 2]
```diff
@@ -12,28 +12,143 @@ import (
...
+			for _, v := range records {
+				if v {
+					totalMetrics++
+				}
+			}
+		}
+	}
+
+	if exportableInstance != expectedExportedInst {
+		t.Errorf("Exported instances got= %d, expected: %d", exportableInstance, expectedExportedInst)
```
Comment (cgrinds) [inline cmd/collectors/zapiperf/zapiperf_test.go:145]: t.Errorf("Exported instances got=%d, expected=%d", exportableInstance, expectedExportedInst)

File: cmd/collectors/zapiperf/zapiperf_test.go [hunk 3]
```diff
@@ -12,28 +12,143 @@ import (
...
+	}
+
+	// Check if the total number of metrics matches the expected value
+	if totalMetrics != expectedMetrics {
+		t.Errorf("Total metrics got= %d, expected: %d", totalMetrics, expectedMetrics)
```
Comment (cgrinds) [inline cmd/collectors/zapiperf/zapiperf_test.go:150]: t.Errorf("Total metrics got=%d, expected=%d", totalMetrics, expectedMetrics)

**Review by cgrinds — APPROVED** [review]

**Review by Hardikl — APPROVED** [review]
looks good

**Pull request merged**